add fallback to fetch endorsers from RPC if sync fails to get that data

dvanderbeek opened the pull request:
[Notion Task](https://www.notion.so/figmentnetworks/Handle-missing-endorsers-8c653d5d04dc438696f68e6957093b6e)

pawelnguyen commented on app/models/tezos/block.rb:
```diff
@@ -32,4 +32,22 @@ def timestamp
+
+    update_columns(endorsers: a)
+
+    return a
```
can we use a bit more descriptive variable names? :)

dvanderbeek commented:
@pawelnguyen @silentlight I updated this per your comments and added a ticket to create a spec for it [here](https://www.notion.so/figmentnetworks/Test-endorsers-fallback-584c3280e17b44fa82c49e1bca74f1e7)

silentlight commented on app/models/tezos/block.rb:
```diff
@@ -32,4 +32,22 @@ def timestamp
+
+    endorsers_from_rpc = []
+    url = Tezos::Rpc.new(chain).url("blocks/#{height}/helpers/endorsing_rights")
+    res = Typhoeus.get(url)
```
Do we have a client for Tezos? Adding this low level Typhoeus call in the model might be problematic in the long run. Using client "service" would be better since we would need to make changes only in one place rather than searching through the code trying to find references.

dvanderbeek replied:
Yeah I can do a little refactoring. I already have another task for adding a test, so I would prefer to add that to the other ticket and get this merged first since it will prevent an issue we have seen in the live indexer.

dvanderbeek commented:
@silentlight @pawelnguyen I updated this to use a separate class for getting Endorsers from the RPC and used that class in both the main sync service and the Block class in case for some reason it doesn't get the results properly during the sync process.

dvanderbeek commented:
it would be great to get this merged and deployed because the indexer keeps stalling out when this issue happens. When that happens Hubble shows stale data until I go in and manually fix the issue so the sync can run properly. This should let it resolve on its own.
